### ISSUE TITLE
X-Axis Ticks and Numeric Labels Overlap

### ISSUE THREAD

**Issue by dcwuser** [oxyplot/oxyplot#1214]
*IF BUG, INCLUDE THIS PART:*

### Steps to reproduce

Create a scatter plot with x-values between ~130 and ~200.

### Expected behaviour

X-axis ticks and numeric scale labels should be cleanly separated and legible.

### Actual behaviour

X-axis ticks and numeric scale labels overlap, which severely impairs legibility, makes graph completely unusable for publication.

Can you also include a screen shot?

![image](https://user-images.githubusercontent.com/20149564/41322049-b47c6614-6e5b-11e8-997a-9137e5945a60.png)

**Comment by dcwuser** [comment]
Sorry, I hadn't thought about the fact that could make a difference. It's in a Xamarin Workbook (https://docs.microsoft.com/en-us/xamarin/tools/workbooks/) running on Windows 10. I don't know what image format is output by OxyPlot in that context. The only OxyPlot NuGet package added is OxyPlot.Core, which perhaps gives you a clue.

I tried to drag-and-drop the workbook into this bug, but github tells me "we don't support that file type". I will mess around and try to get it attached anyway. (I just want github to hold it as a bag of bits, not try to display it, so they shouldn't care about file type.)

**Comment by dcwuser** [comment]
Okay, I put the workbook into a gist: https://gist.github.com/dcwuser/8f11c3766e791903c67bc3660dead52d

**Comment by VisualMelon** [comment]
Judging by https://gist.github.com/abock/20cf5fb709baf61ade1ce65880169253 (which from a position of complete naivety I'm assuming is underneath; I'm struggling to find any documentation (can you link to any?)) it is indeed an SVG it is outputting, which suggests it's the same issue as #549. Just to make sure, is it possible to save the output image to a file and upload it (through a right-click menu or something?)? If it is #549, then it should look correct in e.g. Firefox, but not in e.g. Internet Explorer.

Unfortunately #549 is the result of the OxyPlot SVG renderer using a standard attribute that doesn't have full support everywhere. I've written a PR with a workaround (#1216), but that's unlikely to help your situation unless the authors of the workbook component engage it includes some global flag, which I doubt anybody wants.

Is it possible to use a different version of OxyPlot? I could quickly throw together a version of my fork with the workaround on by default (and without the binary compatibility concerns), and you could try using that instead.

**Comment by dcwuser** [comment]
I'm afraid can't do a direct image export because Xamarin Workbooks doesn't yet support it (https://github.com/Microsoft/workbooks/issues/408). I got the attached image via screenshot. It's not clear to me how Workbooks recognizes that PlotModel can be displayed as an image instead of via the default ToString() rendering; I had actually assumed that Workbooks had special handling for OxyPlot objects because I didn't see any obvious BCL image types exposed by PlotModel and they highlight their OxyPlot support (https://developer.xamarin.com/workbooks/). I assume, though,  that the fact that I am only referencing the Core OxyPlot package, and not the  WPF or WinForms OxyPlot packages, means that you couldn't be producing anything except SVG.

I would be happy to try your modified package. Easiest for me would be if it were published as a pre-release OxyPlot.Core package to NuGet, because Workbooks provides a straightforward way to pull in NuGet packages. I'm not sure if I could force Workbooks to use a DLL on my file system, but I'm willing to try.

(I am curious, btw, why you aren't in a position to test workbooks. Of course, you are under no obligation to de-bug my scenario. But you are clearly willing to put a lot of time and effort into the OxyPlot rendering issues. Workbooks is free, from a reputable source (Xamarin is now MSFT), open-source, and as the only Jupyter Notebooks like environment for .NET, an important scenario for a plotting library.)

**Comment by VisualMelon** [comment]
It looks you can write custom providers; [this one](https://gist.github.com/abock/20cf5fb709baf61ade1ce65880169253) presumably ships with WorkBooks already. It might be possible, as an interim solution, to provide a [base64 encoded png](https://stackoverflow.com/questions/5242319/what-does-this-mean-image-pngbase64) using the PNG exporter as part of a custom provider. 

I'd be surprised the maintainer of the provider hadn't noticed/been notified of this. Perhaps we should nudge him if we make any progress.

> I would be happy to try your modified package...

I don't know the first thing about publishing on NuGet, but I'll see if I can work out how to do that without creating too much confusion.

> why you aren't in a position to test workbook

No suitable workstation which isn't completely locked down, that's all. I use OxyPlot for work, so it gets a disproportionate amount of my attention.

**Comment by dcwuser** [comment]
Yeah, it's special handling that ships with Workbooks (https://github.com/Microsoft/workbooks/blob/ea4144d3da207daf58de51088c8c39a8750bc57c/Agents/Xamarin.Interactive/Representations/ReflectionRepresentationProvider.cs). It hands the PlotModel to OxyPlot.SvgEpxorter.ExportToString.

I'm not a fan of the PNG alternative, since that would loose the re-scalability of the SVG. On the other hand, it looks like it would be very easy for Aaron Bockover (@abock) to call ExportToString with useVerticalTextAlignmentWorkaround = true once your OxyPlot PR (https://github.com/oxyplot/oxyplot/pull/1216) goes through.

**Comment by objorke** [comment]
#1216 has been merged. Is the issue solved?
